indexer/postgres/params.go

```diff
@@ -17,6 +22,9 @@ func (tm *objectIndexer) bindKeyParams(key interface{}) ([]interface{}, []string
 	} else if n == 1 {
 		return tm.bindParams(tm.typ.KeyFields, []interface{}{key})
 	} else {
+		if kc, ok := key.(keyCollection); ok {
+			return tm.bindParams(tm.typ.KeyFields, kc.Keys())
+		}
```

Comment:
We shouldn't need to change anything in indexer here. This breaks the schema specification. Instead we need to make sure that pair, triple, etc in collections decode correctly to what schema expects. This would be part of the codec implementation

Reply:
The SchemaCodec method of PairKeyCodec, etc is what needs to be fixed. I didn't complete the encoding/decoding there properly. Apologies that this wasn't clear.